stocks2.lua

```diff
@@ -0,0 +1,544 @@
+-- a v50 script remake of the older stocks plugin
```

Comment:
We should just name this "stocks", if it's a drop-in replacement and we aren't intending on resurrecting the plugin.

Reply:
I had it named stocks.lua originally but somehow DFHack was bringing up an old stocks plugin description and upon running the command, I would get errors about possible compatibility problems, etc.  For now, I just renamed it to stocks2 since I wasn't sure what the fix was.